### PR TITLE
[test] Remove `output_nicerizer` test feature. NFC

### PR DESCRIPTION
**Pull request by sbc100**
After #22295 there are no more required uses of this feature

### CODE REVIEW

File: test/test_core.py [hunk 1]
```diff
@@ -3086,18 +3077,17 @@ def test_dlfcn_qsort(self):
...
-        lib_handle = dlopen("liblib.so", RTLD_NOW);
+        void* lib_handle = dlopen("liblib.so", RTLD_NOW);
         if (lib_handle == NULL) {
           printf("Could not load lib.\\n");
           return 1;
         }
-        getter_ptr = (CMP_TYPE (*)()) dlsym(lib_handle, "get_cmp");
-        if (getter_ptr == NULL) {
+        cmp_t lib_cmp = (cmp_t)dlsym(lib_handle, "lib_cmp");
+        if (lib_cmp == NULL) {
```
Comment (kripken) [inline test/test_core.py:3086]: These changes look separate, or am I not seeing the connection?

Comment (sbc100) [reply test/test_core.py:3086]: I just simplified the test a little while I was trying to see what it was doing.

Comment (sbc100) [reply test/test_core.py]: Reverted these changes

File: test/test_core.py [hunk 2]
```diff
@@ -5443,15 +5442,11 @@ def test_files(self):
 
     self.do_run_in_out_file_test('test_files.c')
 
-  def test_files_m(self):
-    # Test for Module.stdin etc.
-    # needs to flush stdio streams
-    self.set_setting('EXIT_RUNTIME')
```
Comment (kripken) [inline test/test_core.py:5449]: It sounds like this was testing this option before? Or am I misunderstanding the comment?

Comment (sbc100) [reply test/test_core.py:5449]: Just adding a final newline is enough to flush the stdio streams.  No need for the explicit `EXIT_RUNTIME` here.  I don't think this pertains to the actual test, which is why I removed it.

**Review by kripken — APPROVED** [review]

**Pull request merged**